### PR TITLE
Allow empty replacement with ReplacePathRegex middleware

### LINKED ISSUE #10527: ReplacePathRegex - Allow Empty String for `replacement`
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you expect to see?

Currently using `v2.10.7`, would like to request passing in an empty string for the `replacement` entry for the [ReplacePathRegex](https://doc.traefik.io/traefik/middlewares/http/replacepathregex/) middleware.  At the moment, I see that it requires the `replacement` entry to [not be empty](https://github.com/traefik/traefik/blob/c1ef7429771104e79f2e87b236b21495cb5765f0/pkg/middlewares/replacepathregex/replace_path_regex.go#L55) (with it [trimmed first](https://github.com/traefik/traefik/blob/c1ef7429771104e79f2e87b236b21495cb5765f0/pkg/middlewares/replacepathregex/replace_path_regex.go#L39)).

Main use case for this request is to remove any trailing slashes from the path without having to redirect:
```yml
http:
  middlewares:
    remove-trailing-slash-from-path:
      replacepathregex:
        regex: "/$"
        replacement: ""
```

Some points around the request/confusion:
- In the current documentation for the middleware, the [field `replacement`](https://doc.traefik.io/traefik/middlewares/http/replacepathregex/#replacement) doesn't mention that it needs to be non-empty or white space
- When testing out the values in [Regex101](https://regex101.com/r/Px222C/1) and in [Go Playground](https://go.dev/play/p/_BhJmJcxUv8) (noted in the `Tips` section in the docs), the replacement works as intended (i.e., trailing slash removed)
- Looks like [`keepTrailingSlash` was deprecated some time ago](https://github.com/traefik/traefik/issues/4247) + haven't found any answers or solutions in the [Community Boards](https://community.traefik.io/search?q=trailing%20slash)

Any Traefik-ers have suggestions around how to do this with the latest version of Traefik, it'll be greatly appreciated, would like this to be considered as a feature request, thanks!

### PR DESCRIPTION
**Pull request by rtribotte**
### What does this PR do?

This PR modifies the ReplacePathRegex middleware to allow empty replacement. 

### Motivation

Fixes #10527

### More

- [x] Added/updated tests
- [ ] Added/updated documentation

### CODE REVIEW

**Review by kevinpollet — APPROVED** [review]

File: pkg/middlewares/replacepathregex/replace_path_regex_test.go [hunk 1]
```diff
@@ -44,6 +44,17 @@ func TestReplacePathRegex(t *testing.T) {
 			expectedRawPath: "/who-am-i/and/who-am-i",
 			expectedHeader:  "/whoami/and/whoami",
 		},
+		{
```
Comment (danhlee329) [inline pkg/middlewares/replacepathregex/replace_path_regex_test.go:47]: Can we also add tests with whitespace to confirm that this is being trimmed?

Comment (danhlee329) [reply pkg/middlewares/replacepathregex/replace_path_regex_test.go:47]: awesome, thank you!

**Review by ldez — APPROVED** [review]
LGTM
